material_studies/rally/lib/finance.dart

```diff
@@ -0,0 +1,333 @@
+                            .body1
+                            .copyWith(fontSize: 16.0),
+                      ),
+                      Text(
+                        subtitle,
+                        style: Theme.of(context)
+                            .textTheme
+                            .body1
+                            .copyWith(color: RallyColors.gray60),
+                      ),
+                    ],
+                  ),
+                  Spacer(),
+                  Text(
+                    '\$ ' + Formatters.usd.format(amount),
+                    style: Theme.of(context)
+                        .textTheme
+                        .body2
+                        .copyWith(fontSize: 20.0, color: RallyColors.gray),
+                  ),
+                  SizedBox(width: 32.0, child: suffix),
+                ],
+              ),
+            ),
+            Divider(
+              height: 1,
+              indent: 16,
+              endIndent: 16,
+              color: Color(0xAA282828),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+/// Data model for [FinancialEntityCategoryView].
+class FinancialEntityCategoryModel {
+  final Color indicatorColor;
+  final double indicatorFraction;
+  final String title;
+  final String subtitle;
+  final double usdAmount;
+  final Widget suffix;
+
+  const FinancialEntityCategoryModel(
+    this.indicatorColor,
+    this.indicatorFraction,
+    this.title,
+    this.subtitle,
+    this.usdAmount,
+    this.suffix,
+  );
+}
+
+FinancialEntityCategoryView fromAccountItem(AccountData model, int i) {
+  return FinancialEntityCategoryView(
+    suffix: Icon(Icons.chevron_right, color: Colors.grey),
+    title: model.name,
+    subtitle: '• • • • • • ${model.accountNumber.substring(6)}',
+    indicatorColor: RallyColors.accountColor(i),
+    indicatorFraction: 1.0,
+    amount: model.primaryAmount,
+  );
+}
+
+FinancialEntityCategoryView fromBillItem(BillData model, int i) {
+  return FinancialEntityCategoryView(
+    suffix: Icon(Icons.chevron_right, color: Colors.grey),
+    title: model.name,
+    subtitle: model.dueDate,
+    indicatorColor: RallyColors.billColor(i),
+    indicatorFraction: 1.0,
+    amount: model.primaryAmount,
+  );
+}
+
+FinancialEntityCategoryView fromBudgetItem(
+    BudgetData item, int i, BuildContext context) {
+  return FinancialEntityCategoryView(
+    suffix: Text(' LEFT',
+        style: Theme.of(context)
+            .textTheme
+            .body1
+            .copyWith(color: RallyColors.gray60, fontSize: 10.0)),
+    title: item.name,
+    subtitle: Formatters.usdWithSign.format(item.amountUsed) +
+        ' / ' +
+        Formatters.usdWithSign.format(item.primaryAmount),
+    indicatorColor: RallyColors.budgetColor(i),
+    indicatorFraction: item.amountUsed / item.primaryAmount,
+    amount: item.primaryAmount - item.amountUsed,
+  );
+}
+
+List<FinancialEntityCategoryView> fromAccountDataList(List<AccountData> items) {
```

Comment:
Nice work to get these into top-level methods, but without the old class name, the method names no longer make sense. Something like `buildAccountDataListViews` would be more descriptive.

Reply:
Done